src/main/java/Controller/Engine/Stockfish.java

```diff
@@ -0,0 +1,78 @@
+package Controller.Engine;
+
+import Model.Game;
+import Model.Pieces.PieceType;
+import Model.Position;
+
+import java.io.*;
+import java.util.Objects;
+
+import static java.util.Objects.isNull;
+
+public class Stockfish implements Runnable{
+
+    private Process process;
+    private String command;
+    private Game game;
+
+    public Stockfish(String fen, Game game){
+        this.game = game;
+        ProcessBuilder builder = new ProcessBuilder();
+        builder.command("stockfish_14.1_win_x64_avx2.exe");
+        builder.redirectErrorStream(true);
+        try {
+            process = builder.start();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        command = "position fen " + fen +"\n";
+    }
+
+    @Override
+    public void run(){
+        StockFishOutput stockFishOutput = getStockfishMove();
+        if(isNull(stockFishOutput.getPieceType())){
+            game.makeAMove(stockFishOutput.getPreviousPosition(),stockFishOutput.getNewPosition(), true);
+        } else{
+            game.makeAMove(stockFishOutput.getPreviousPosition(),stockFishOutput.getNewPosition(), true);
+            game.promotionPieceDecision(stockFishOutput.getPieceType());
+        }
+    }
+
+
+    public StockFishOutput getStockfishMove() {
+        try {
+            BufferedReader r = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            BufferedWriter w = new BufferedWriter(new OutputStreamWriter(process.getOutputStream()));
+
+            w.write(command);
+            w.write("go movetime 500\n");
+            w.flush();
+
+            String line;
+            while (true) {
```

Comment:
so confused by this line - it never loops once?

Reply:
It does loop. `r.readLine()` reads the next line in the `BufferedReader`, so need it to go through all the lines. Without it it would just read the first line